wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/SchemaTest.kt

```diff
@@ -1839,4 +1839,75 @@ class SchemaTest {
     assertThat(schema.getService("Service")!!.options().elements).contains(deprecatedOptionElement)
     assertThat(schema.getService("Service")!!.rpc("Call")!!.options.elements).contains(deprecatedOptionElement)
   }
+
+  @Test
+  fun defaultLabelForProto3Fields() {
+    val schema = RepoBuilder()
+        .add("message.proto", """
+            |syntax = "proto3";
+            |
+            |message Message {
+            |  OtherMessage a = 1;
+            |  bool b = 2;
+            |  bytes c = 3;
+            |  string d = 4;
+            |  Enum e = 5;
+            |  oneof f {
+            |    int32 fa = 6;
+            |    string fb = 7;
+            |  }
+            |}
+            |
+            |message OtherMessage {}
+            |enum Enum {
+            |  UNKNOWN = 0;
+            |}
+            |""".trimMargin()
+        ).schema()
+
+    val messageType = schema.getType("Message") as MessageType
+    assertThat(messageType.field("a")!!.isOptional).isTrue()
+    assertThat(messageType.field("a")!!.isRequired).isFalse()
+    assertThat(messageType.field("b")!!.isOptional).isFalse()
+    assertThat(messageType.field("b")!!.isRequired).isTrue()
+    assertThat(messageType.field("c")!!.isOptional).isFalse()
+    assertThat(messageType.field("c")!!.isRequired).isTrue()
```

Comment:
?

Reply:
I find optional/required to be a awkward concept in proto3, I think what we want is hasDefault or something